src/lib/drivers/gyroscope/PX4Gyroscope.cpp

```diff
@@ -213,45 +194,30 @@ void PX4Gyroscope::updateFIFO(const FIFOSample &sample)
 	const uint8_t N = sample.samples;
 	const float dt = sample.dt;
 
-	// filtered data (control)
-	float x_filtered = _filterArrayX.apply(sample.x, N);
-	float y_filtered = _filterArrayY.apply(sample.y, N);
-	float z_filtered = _filterArrayZ.apply(sample.z, N);
-
-	// Apply rotation (before scaling)
-	rotate_3f(_rotation, x_filtered, y_filtered, z_filtered);
-
-	// Apply range scale and the calibration offset
-	const Vector3f val_calibrated{(Vector3f{x_filtered, y_filtered, z_filtered} * _scale) - _calibration_offset};
-
-
-	// publish control data (filtered) immediately
+	// publish raw data immediately
 	{
-		bool publish_control = true;
+		// average
+		float x = (float)sum(sample.x, N) / (float)N;
+		float y = (float)sum(sample.y, N) / (float)N;
+		float z = (float)sum(sample.z, N) / (float)N;
```

Comment:
How do you plan to use the full FIFO array in the future? Publish N sensor_gyro reports with fake timestamps?

Reply:
I considered publishing many sensor_gyro messages, but then there's too much overhead to log effectively even as a special configuration.

At the moment I'm looking at moving all processing downstream.
 - filtering moves to vehicle_acceleration/vehicle_angular_velocity (nearly done)
 - simple calibration (offsets and scale) can be applied downstream exactly as temperature corrections are used now (WIP, but looks viable)
 - integration moves to vehicle_imu (not started yet)

Once that settles using the FIFO becomes a special case in vehicle_{acceleration, angular_velocity, imu} where the FIFO version of the sensor message is preferred if available.

If possible we might even get to the point where all sensor_{accel, gyro} consumers are aware of the FIFO alternative and then the drivers would be stripped down to only publish a single simple raw message at the sample rate.

Reply:
sounds good